nano/node/wallet.cpp

```diff
@@ -1512,6 +1534,67 @@ void nano::wallets::reload ()
 	}
 }
 
+void nano::wallets::do_work_regeneration ()
+{
+	std::unique_lock<std::mutex> regeneration_lock (difficulty_mutex);
+	while (!stopped)
+	{
+		if (!difficulty_reque.empty ())
+		{
+			auto const first (difficulty_reque.begin ());
+			auto now (std::chrono::steady_clock::now ());
+			auto queued (first->first);
+			auto block (first->second);
+			std::shared_ptr<nano::block> block_l (block);
+			bool confirmed (false);
+			difficulty_reque.erase (first);
+			if ((now - queued) >= node.config.work_recalc_inverval)
+			{
+				std::unique_lock<std::mutex> lock (node.active.mutex);
+				auto existing (node.active.roots.find (block->qualified_root ()));
+				if (node.active.roots.end () != existing)
```

Comment:
Probably else condition should trigger `confirmed = true;` or at least check with ledger.block_confirmed ()

Reply:
else now creates block transaction and sets confirmed to `ledger.block_confirmed(transaction,block->hash())`